### ISSUE TITLE
[FEATURE REQUEST] 3003 Need a way to refresh next_job_fire_time in the pillar schedule

### ISSUE THREAD

**Issue by ninoY25** [saltstack/salt#60300]
**Is your feature request related to a problem? Please describe.**
If I change the minion system time frontwards, and run a schedule job once. Then adjust the time backwards, because the next fire time of schedule job will be some time in the future, the schedule job will be running only after some future time I changed at the first time. 
e.g. 
If I change the minion system time to 2022-06-02, and run a schedule job A. And the interval of schedule job A is one day.
Therefore, next fire time of job A will be 2022-06-03. Then, if I change system time back to 2021-06-02, I have to wait until 2022-06-03 for next scheduled job A running. If I want to refresh the next fire time value of the job A, I have to restart the minion because there is no way to refresh it manually.

**Describe the solution you'd like**
I want to refresh the pillar schedule when I reload the schedule in the minion. `schedule.reload` function should refresh the pillar schedule, and should get rid of the _next_fire_time in the pillar schedule variable. 
In the 3002.6 or previous version, I can use saltutil.refresh_pillar/schedule.reload to reload all schedule in the pillar. Because the PR https://github.com/saltstack/salt/pull/59104, refresh_pillar only refresh the new schedule in the pillar. It keeps the old schedule with their old attributes like ` _next_fire_time` in the opts['pillar'] dict. I think these attributes should be popped when the schedule is reloaded.

**Comment by garethgreenaway** [comment]
@ninoY25 Thanks for the report. I've read through this one a couple times and I'm struggling to think of a use case when you would want to adjust the time of the system that far into the future and still scheduled jobs work as expected.  The PR that you mentioned fixed a bug when `refresh_pillar` was run and would delete all the elements in Pillar include `_next_fire_time`, thus removing all history of when a job was last run and should be run again.  The closest thing we have today that will accomplish what you're trying to do is to restart the Salt Minion process, this will clear everything out.

**Comment by ninoY25** [comment]
@garethgreenaway I think the schedule.reload should cover the function to reload the pillar schedule like reloading the file schedule. But currently, the schedule.reload function just use the modified refresh pillar function, and it will skip reload the existing pillar schedule. There should be some ways to refresh the pillar schedule. For example, we can create the pillar_schedule_reload method in minion class and triggered by the event pillar_schedule_reload.

**Comment by garethgreenaway** [comment]
Closing this due to age and the old version of Salt.
Can you retest this with the latest release of Salt 3006.2 and if this is still an issue, please open a new issue.
